docs/guides/custom-functions.md

```diff
@@ -145,6 +145,68 @@ rules:
 
 Spectral would look for functions in a `my-functions` directory now.
 
+## Returning multiple results
+
+Spectral results are automatically deduplicated when they bear the same `code` and target the same `path`.
```

Comment:
Many functions will return a single message, but its possible for a function to return multiple. 

For example, if a rule is created to make sure something is unique, it could either:

a) return a single error for the entire array which lists offending values in a comma separated list
b) return a single error for the array value which contains the first offending non-unique item
c) return multiple errors for each duplicate value located

How exactly you chose to implement messages depends on the rule at hand and probably personal preference too.

It's worth keeping in mind, Spectral will attempt to deduplicate messages based on....

Reply:
Fixed